components/homme/src/share/derivative_mod_base.F90

```diff
@@ -1241,6 +1242,54 @@ function vlaplace_sphere_wk_contra(v,deriv,elem,var_coef,nu_ratio) result(laplac
     enddo
   end function vlaplace_sphere_wk_contra
 
+!DIR$ ATTRIBUTES FORCEINLINE :: second_order_findiff
+  function second_order_findiff(u1, u2, u3, zeta1, zeta3) result(du_dzeta)
+        real(kind=real_kind), intent(in) :: u1, u2, u3, zeta1, zeta3
```

Comment:
So this computes 2nd order deriv given u1=u(x1),u2=u(x2),u3=u(x3), with zeta1=|x2-x1| and zeta3=|x3-x2|? Might be worth adding a comment explaining this.

Reply:
The derivation can be seen mid-way through this lab notebook post: https://open-lab-notebook.glitch.me/posts/E3SM/potential_vorticity/. I'll suggest an explanatory comment tomorrow.

Reply:
A comment in the code here would be good @OkayHughes @jhollowed

Reply:
I suggest the following comment:
`! This method computes second-order finite difference approximation to a field u at irregularly spaced points.
! let u_0, u_1, u_2 be evaluations of a quantity at coordinates z_0, z_1, z_2, respectively.
! Let a_1, a_2, a_3 be coeffients of a polynomial $f(z)$ that approximates u, 
! such that u_i = a_0 (z_i-z_1)^2 + a_1 (z_i-z_1) + a_2.
! This is a second order polynomial passing through the points (z_i, u_i).
! A second order approximation to du/dz is then f(z_i) = a_1. 
! Noting that u_1 = a_0 * 0^2 + a_1 * 0 + a_2, a_2 = u_1. 
! Define zeta_i = z_i - z_1.
! The other two constraints f(z_0) = u_0 and f(z_2) = u_2 give a 2x2 system of equations
! [zeta_0^2, zeta_0 ; zeta_2^2, zeta_2 ][a_0; a_1] = [u_0-u_1; u_2-u_1].
! We use the standard closed-form way of solving a 2x2 system.
! This method returns the value a_1 computed when this system is solved.`